packages/protobuf/src/message.ts

```diff
@@ -132,6 +132,25 @@ export class Message<T extends Message<T> = AnyMessage> {
     return Object.getPrototypeOf(this).constructor;
   }
 }
+/**
+ * isMessage checks if the given value is a Message<T> or not.
+ */
+/* eslint-disable @typescript-eslint/strict-boolean-expressions,@typescript-eslint/no-unsafe-member-access,@typescript-eslint/no-explicit-any */
+export function isMessage<T extends Message<T>>(value: any): value is T {
+  return (
+    !!value &&
+    typeof value.equals === "function" &&
+    typeof value.clone === "function" &&
+    typeof value.fromBinary === "function" &&
+    typeof value.fromJson === "function" &&
+    typeof value.fromJsonString === "function" &&
+    typeof value.toBinary === "function" &&
+    typeof value.toJson === "function" &&
+    typeof value.toJsonString === "function" &&
+    typeof value.getType === "function"
+  );
+}
```

Comment:
Hmm... I don't think this is the ideal approach to this. I'd suggest to instead implement a generic ttype guard on the MessageType instead and, in there, only loosely check that the given Mesasge has a getType function and that it returns the same message type (typeName comparison)

```typescript
is(message: unknown): message is Message<ThisMessageType> {
  // Check that `message` has a `getType` function and that that returns
  // a `MessageType` with `typeName === this.typeName`
}
```